overcast/runner/__init__.py

```diff
@@ -400,6 +443,24 @@ def provision_step(self, details):
         else:
             userdata = None
 
+        pending_nodes = set()
+        done          = set()
+
+        if details.get('retry-delay', False):
+            retry_delay = utils.parse_time(details['retry-delay'])
+        else:
+            retry_delay = 5
+
+        if details.get('timeout', False):
+            provision_timeout = utils.parse_time(details['timeout'])
+            deadline =  time.time() + provision_timeout
+        else:
+            provision_timeout = None
+            deadline = None
```

Comment:
Some commands need to be retried.  They're simply not expected to work the first time, so we keep running them until they work. We feel confident that they'll eventually work, but know that it will take a while.

Other commands are different. If they don't work, we know they'll never work.

The point is, different commands will have different retry settings. I just don't see the same being true for provision steps. Either you want to retry a few times (because you're more interested in getting things up and running) or you don't (because you're more interested exercising the cloud than in the actual resources you're creating). So, I see this as a global setting and I see it as a setting that the user defines when invoking overcast (i.e. on the command line), rather than a setting that belongs in the yaml file.